go/enclave/enclave.go

```diff
@@ -600,6 +609,160 @@ func (e *enclaveImpl) GetLogs(encryptedParams common.EncryptedParamsGetLogs) (co
 	return encryptedLogs, nil
 }
 
+// DoEstimateGas returns the estimation of minimum gas required to execute transaction
+// This is a copy of https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1055
+func (e *enclaveImpl) DoEstimateGas(args *gethapi.TransactionArgs, blockNr gethrpc.BlockNumber, gasCap uint64) (hexutil.Uint64, error) { //nolint: gocognit
```

Comment:
I think if you can find a way to not ignore the complexity and nested-if lint failures, then this method will be more easy to follow.

Reply:
I prefer to no rewrite the copy of the geth code. At least not until we do the gas mechanics work.
Will add that to the comment 👍